sos/report/plugins/block.py

```diff
@@ -35,6 +35,9 @@ def setup(self):
             "ls -lanR /sys/block",
             "lsblk -O -P",
             "losetup -a",
+            "ls -l /dev/disk/by-path/",
+            "ls -l /dev/disk/by-id/",
+            "ls -l /dev/disk/by-uuid/",
```

Comment:
As far as I know we already capture this a bit before line 35, via 'ls -lanR /dev'. What's the reason to have this data captured again in three new files?

Reply:
I don't think it's covered by line 35 - at least not obviously

`ls -lanR /sys/block`:
```
lrwxrwxrwx.  1 0 0 0 Jul 28 18:53 dm-0 -> ../devices/virtual/block/dm-0
lrwxrwxrwx.  1 0 0 0 Jul 28 11:58 dm-1 -> ../devices/virtual/block/dm-1
lrwxrwxrwx.  1 0 0 0 Jul 28 11:58 dm-2 -> ../devices/virtual/block/dm-2
lrwxrwxrwx.  1 0 0 0 Jul 28 18:53 nvme0n1 -> ../devices/pci0000:00/0000:00:01.1/0000:01:00.0/nvme/nvme0/nvme0n1
lrwxrwxrwx.  1 0 0 0 Jul 28 18:53 sda -> ../devices/pci0000:00/0000:00:01.3/0000:02:00.1/ata1/host0/target0:0:0/0:0:0:0/block/sda
lrwxrwxrwx.  1 0 0 0 Jul 28 18:53 sdb -> ../devices/pci0000:00/0000:00:01.3/0000:02:00.1/ata2/host1/target1:0:0/1:0:0:0/block/sdb
lrwxrwxrwx.  1 0 0 0 Jul 28 11:58 sr0 -> ../devices/pci0000:00/0000:00:01.3/0000:02:00.1/ata5/host4/target4:0:0/4:0:0:0/block/sr0
lrwxrwxrwx.  1 0 0 0 Jul 28 11:58 zram0 -> ../devices/virtual/block/zram0
```

`ls -lanR /dev/disk/by-uuid`:
```
lrwxrwxrwx. 1 0 0  10 Aug  1 09:59 042ED13D2ED12906 -> ../../sda2
lrwxrwxrwx. 1 0 0  10 Aug  1 09:59 0d771bef-e2ab-4be8-b1f1-2084b89d02ed -> ../../dm-2
lrwxrwxrwx. 1 0 0  10 Aug  1 09:59 A05C66B85C6688BC -> ../../sdb1
lrwxrwxrwx. 1 0 0  10 Aug  1 09:59 ad056e96-42d1-4f22-981e-d30860716902 -> ../../dm-0
lrwxrwxrwx. 1 0 0  10 Aug  1 09:59 B4A26C4EA26C16E0 -> ../../sda1
lrwxrwxrwx. 1 0 0  15 Aug  1 09:59 D901-B04B -> ../../nvme0n1p1
lrwxrwxrwx. 1 0 0  10 Aug  1 09:59 f48b7f19-4d48-44ee-a762-3f2c60bb2216 -> ../../dm-1
```

Reply:
It's done in line 34.  This is what we have in a RHEL 8 sos for example for by-id:

/dev/disk/by-id:
total 0
drwxr-xr-x. 2 0 0 440 Jul 27 05:03 .
drwxr-xr-x. 6 0 0 120 Jul 27 05:03 ..
lrwxrwxrwx. 1 0 0   9 Jul 27 05:03 ata-QEMU_DVD-ROM_QM00001 -> ../../sr0
lrwxrwxrwx. 1 0 0  10 Jul 27 05:03 dm-name-mpathcs -> ../../dm-0
lrwxrwxrwx. 1 0 0  10 Jul 27 05:03 dm-name-mpathct -> ../../dm-1
lrwxrwxrwx. 1 0 0  10 Jul 27 05:03 dm-name-test_wipefs-swap -> ../../dm-3
lrwxrwxrwx. 1 0 0  10 Jul 27 05:03 dm-name-test_wipefs-test2 -> ../../dm-2
lrwxrwxrwx. 1 0 0  10 Jul 27 05:03 dm-uuid-LVM-uKK4ct8wcZ3mQwyrr1ZvsvZL1hbbzcuXH05pHuJWfGQqLgjUZKUhXWjIwtLCvtwZ -> ../../dm-2
lrwxrwxrwx. 1 0 0  10 Jul 27 05:03 dm-uuid-LVM-uKK4ct8wcZ3mQwyrr1ZvsvZL1hbbzcuXrDZLcuq4Ru2ZGVmNaYV9ISgU4tNSOIc1 -> ../../dm-3
lrwxrwxrwx. 1 0 0  10 Jul 27 05:03 dm-uuid-mpath-0QEMU_QEMU_HARDDISK_drive-scsi0-0-0-1 -> ../../dm-0
lrwxrwxrwx. 1 0 0  10 Jul 27 05:03 dm-uuid-mpath-0QEMU_QEMU_HARDDISK_drive-scsi0-0-0-2 -> ../../dm-1
lrwxrwxrwx. 1 0 0  10 Jul 27 05:03 lvm-pv-uuid-IF4HsJ-lgMY-d2Bt-BWKZ-u4Sn-TfU8-2bQDD7 -> ../../dm-0
lrwxrwxrwx. 1 0 0  11 Jul 27 05:03 md-name-localhost.localdomain:boot -> ../../md126
lrwxrwxrwx. 1 0 0  11 Jul 27 05:03 md-name-localhost.localdomain:root -> ../../md127
lrwxrwxrwx. 1 0 0  11 Jul 27 05:03 md-name-localhost.localdomain:swap -> ../../md125
lrwxrwxrwx. 1 0 0  11 Jul 27 05:03 md-uuid-04a8288b:8573ac4f:db82e5c4:fe922ff9 -> ../../md127
lrwxrwxrwx. 1 0 0  11 Jul 27 05:03 md-uuid-1cbf6e0e:366f157b:13bf75a4:34001726 -> ../../md126
lrwxrwxrwx. 1 0 0  11 Jul 27 05:03 md-uuid-2c4aaa2d:5733d8b3:2615507a:78aca107 -> ../../md125
lrwxrwxrwx. 1 0 0  10 Jul 27 05:03 scsi-0QEMU_QEMU_HARDDISK_drive-scsi0-0-0-1 -> ../../dm-0
lrwxrwxrwx. 1 0 0  10 Jul 27 05:03 scsi-0QEMU_QEMU_HARDDISK_drive-scsi0-0-0-2 -> ../../dm-1
lrwxrwxrwx. 1 0 0  10 Jul 27 05:03 wwn-0xQEMU_QEMU_HARDDISK_drive-scsi0-0-0-1 -> ../../dm-0
lrwxrwxrwx. 1 0 0  10 Jul 27 05:03 wwn-0xQEMU_QEMU_HARDDISK_drive-scsi0-0-0-2 -> ../../dm-1

These directories should not have any sub directory, so a -R shouldn't be necessary.

Reply:
Ah, nice catch. Yeah, if this is already in another collection we don't need to duplicate it.

Reply:
Hi, Thanks for the review,  
As per the above discussion since we are already collecting  following data in 'ls -lanR /dev',  we will not be needing this patch

ls -l /dev/disk/by-path/",
"ls -l /dev/disk/by-id/",
"ls -l /dev/disk/by-uuid/",